hubtools/src/caboose.rs

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::mem;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CabooseError {
+    #[error("error reading caboose: {0:?}")]
+    TlvcReadError(tlvc::TlvcReadError),
+
+    #[error("caboose missing expected tag {tag:?}")]
+    MissingTag { tag: [u8; 4] },
+}
+
+#[derive(Debug, Clone)]
+pub struct Caboose {
+    raw: Vec<u8>,
+}
+
+impl Caboose {
+    pub(crate) fn new(raw: Vec<u8>) -> Self {
+        Self { raw }
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
+        &self.raw
+    }
+
+    pub fn board(&self) -> Result<&[u8], CabooseError> {
```

Comment:
Nit: seems like we could write a more general `get_tag(&self, tag: [u8; 4])`, then simply

```rust
pub fn board(&self) -> Result<&[u8], CabooseError> {
    self.get_tag(tags::BORD)
}
```

Reply:
Good call, and at that point it's trivial to go ahead and add methods for the other known tags. Changed/added in [0a491f0](https://github.com/oxidecomputer/hubtools/pull/7/commits/0a491f0c176788011ba879821afcf041692ff9b5)